android/app/src/main/kotlin/com/jabook/app/jabook/audio/MediaCodecChapterSignalExtractor.kt

```
@@ -0,0 +1,200 @@
+            }
+
+        private fun findAudioTrackIndex(extractor: MediaExtractor): Int? {
+            for (i in 0 until extractor.trackCount) {
+                val format = extractor.getTrackFormat(i)
+                val mime = format.getString(MediaFormat.KEY_MIME) ?: continue
+                if (mime.startsWith("audio/")) return i
+            }
+            return null
+        }
+
+        private fun decodeWindow(
```

detekt (1.23.8)

[warning] The function decodeWindow appears to be too complex based on Cognitive Complexity (complexity: 27). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)